### ISSUE TITLE
Exception in module triggers 302 redirect to Default.aspx instead off showing Error 500 page

### ISSUE THREAD

**Issue by Swimburger** [dnnsoftware/Dnn.Platform#2364]
## Description of bug
In the site settings of our DotNetNuke installation, we set the Default Page for "500 Error Page" as you can see below. 
![Screenshot DNN Settings](https://i.stack.imgur.com/ws7OU.png)
After setting this, we were expecting to see the "500 Error Page" with HTTP status code 500 when an error occurs. Instead we're still redirected the the "Default.aspx?tabid=..." page with a HTTP 302 status code, followed by HTTP 200 code.

This could cause search engines to index the Default.aspx page which we don't want (it has the robots meta tag set to "Index, Follow".
When an exception occurs, the set "500 Error Page" should be shown, with HTTP Status 500 .

## Steps to reproduce
List the steps to reproduce the behavior:
1. Create a new DNN page "500"
2. Navigate to site settings > Site Behavior > Default Page
3. Set "500 Error Page" to the DNN page "500" created earlier
4. Navigate to a page containing a module that throws any exception. See below for an example module.

Example exception throwing module
```
public partial class TriggerError500 : PortalModuleBase 
{ 
  protected void Page_Load(object sender, EventArgs e) 
  { 
    if(UserId == -1) 
    { 
      throw new NotImplementedException(); 
    } 
  } 
}
```

## Current result
302 redirect to Default.aspx page which returns HTTP status 200.

## Expected result
DNN Page "500" is returned without redirect and with HTTP status code 500.

## Affected version

* [x] 9.02.00 (366)
Other versions not tested.

**Comment by valadas** [comment]
This looks to me to be by design, the label states to which page to redirect users if a 500 error occurs, so the way it works now, is fine if that is what you want. If that is not what you want, then leaving it to default is one option and adding a custom IIS rule is another option. I am closing this issue for that reason and because most discussions about it happened outside if here.

**Comment by Swimburger** [comment]
It's been so long since I submitted this issue, I forgot the details. But roughly I recall this is not just one issue.
1. the 500 settings just doesn't do anything
2. the default.aspx issue with the wrong HTTP status codes.

We weren't able to resolve either, even with the help of the community.
We ended up to the decision to either fix the bugs ourselves in the CMS source code and recompile it ourselves, or accept that this part will malfunction. We did the latter.
